src/libstd/sys/unix/thread.rs

```diff
@@ -65,13 +65,20 @@ impl Thread {
             }
         };
 
-        let ret = libc::pthread_create(&mut native, &attr, thread_start, &*p as *const _ as *mut _);
+        let ret = libc::pthread_create(
+            &mut native,
+            &attr,
+            thread_start,
+            &mut *p as &mut Box<dyn FnOnce()> as *const _ as *mut _,
```

Comment:
```suggestion
            &mut *p as &mut Box<dyn FnOnce()> as *mut _ as *mut _,
```
This should work. The first cast is ref-to-raw but keeps the pointee type; the second changes the pointee.